common/persistence/pinot/pinotVisibilityStore.go

```diff
@@ -749,14 +776,14 @@ func (v *pinotVisibilityStore) getCountWorkflowExecutionsQuery(tableName string,
 	return query.String()
 }
 
-func (v *pinotVisibilityStore) getListWorkflowExecutionsByQueryQuery(tableName string, request *p.ListWorkflowExecutionsByQueryRequest) string {
+func (v *pinotVisibilityStore) getListWorkflowExecutionsByQueryQuery(tableName string, request *p.ListWorkflowExecutionsByQueryRequest) (string, error) {
 	if request == nil {
-		return ""
+		return "", nil
 	}
 
 	token, err := pnt.GetNextPageToken(request.NextPageToken)
 	if err != nil {
-		panic(fmt.Sprintf("deserialize next page token error: %s", err))
+		return "", fmt.Errorf("next page token: %w", err)
```

Comment:
Nit: What about logging the query just before returning the error, else that info will be lost?

Reply:
Sure, added log in the outer method that calls this get query method.